aptos-move/framework/aptos-framework/sources/configs/staking_config.move

```diff
@@ -110,8 +157,39 @@ module aptos_framework::staking_config {
     }
 
     /// Return the reward rate.
-    public fun get_reward_rate(config: &StakingConfig): (u64, u64) {
-        (config.rewards_rate, config.rewards_rate_denominator)
+    public fun get_reward_rate(_config: &StakingConfig): (u64, u64) {
+        assert!(false, EDEPRECATED_FUNCTION);
+        (0, 1)
+    }
+
+    public fun get_epoch_reward_rate(epoch_duration: u64): (u64, u64) acquires StakingRewardsConfig {
+        let staking_rewards_config = borrow_global<StakingRewardsConfig>(@aptos_framework);
+        assert!(staking_rewards_config.year_in_micros > 0, EINVALID_TIME_PERIOD);
+        if (epoch_duration > 0) {
+            (staking_rewards_config.yearly_rewards_rate * epoch_duration / staking_rewards_config.year_in_micros, staking_rewards_config.rewards_rate_denominator)
+        } else {
+            (0, 1)
+        }
+    }
+
+    public fun check_and_autodecrease_rewards_rate(current_time: u64) acquires StakingRewardsConfig {
```

Comment:
Two questions here:
1. Is this intentionally public so anyone can call this to update the rewards rate (no malicious behavior possible here since it's controlled based on last decrease time).
2. What's the benefit of saving the new rewards rate vs just computing it on the fly (initial_reward_time - (current_time - initial time) / year_in_secs * decrease_rate)?

Reply:
1. not intentionally public, what should I make it? friend?
2. three reasons:
  - I don't think we have "initial time". 
  - if at some point we want to change decrease rate, do we want it to retroactrively affect things (doubt it?) or only in the future? 
  - it is really an exponential formula ( * 0.985 every year), not a linear one, though we can make it be linear one instead